staging/src/k8s.io/client-go/tools/remotecommand/websocket.go

```diff
@@ -174,11 +175,23 @@ func (e *wsStreamExecutor) StreamWithContext(ctx context.Context, options Stream
 	}
 }
 
+// IsUpgradeFailure returns true if the passed error is (or wrapped error contains)
+// the UpgradeFailureError.
+func IsUpgradeFailure(err error) bool {
```

Comment:
do you may need this later for port forward?
if that is the case is it better to make thus function available in other place?

Reply:
Yes. I think we'll need it for `PortForward`. Should I refactor now or during the `PortForward` PR?

Reply:
I'm fine either way

Reply:
Moved the `UpgradeFailureError` and `IsUpgradeFailure()` function to `apimachinery/pkg/util/httpstream`.